syncd/syncd_saiswitch.cpp

```diff
@@ -550,55 +539,102 @@ void SaiSwitch::collectCounters(
 
-        return;
+            sai_status_t status = sai_metadata_sai_port_api->get_port_stats(
+                    port_rid,
+                    countersSize,
+                    m_supported_counters.data(),
+                    counters.data());
+
+            if (status != SAI_STATUS_SUCCESS)
+            {
+                SWSS_LOG_ERROR("failed to collect counters for port RID %s: %s",
+                        sai_serialize_object_id(port_rid).c_str(),
+                        sai_serialize_status(status).c_str());
+
+                continue;
+            }
+
+            sai_object_id_t vid = translate_rid_to_vid(port_rid, m_switch_vid);
+
+            std::vector<swss::FieldValueTuple> values;
+
+            for (size_t idx = 0; idx < counters.size(); idx++)
+            {
+                const std::string &field = sai_serialize_port_stat(m_supported_counters[idx]);
+                const std::string &value = std::to_string(counters[idx]);
+                values.push_back( swss::FieldValueTuple(field, value) );
+            }
+
+            countersTable.set(strPortId, values, "");
+        }
+
+        collectQueueCounters(port_rid, countersTable);
     }
+}
 
-    uint32_t countersSize = (uint32_t)m_supported_counters.size();
+void SaiSwitch::collectQueueCounters(
+        _In_ sai_object_id_t port_rid,
+        _In_ swss::Table &countersTable) const
+{
+    SWSS_LOG_ENTER();
 
+    uint32_t countersSize = (uint32_t)m_supported_queue_counters.size();
     std::vector<uint64_t> counters;
-
     counters.resize(countersSize);
 
-    auto ports = saiGetPortList();
+    if (m_queue_rid_map.find(port_rid) == m_queue_rid_map.end())
+    {
+        SWSS_LOG_ERROR("failed to collect queues counters for port RID %s",
+                sai_serialize_object_id(port_rid).c_str());
 
-    for (auto &port_rid: ports)
+        return;
+    }
+
+    const std::vector<sai_object_id_t>& queue_list = m_queue_rid_map.at(port_rid);
+
+    for (auto &queue_rid: queue_list)
     {
-        sai_status_t status = sai_metadata_sai_port_api->get_port_stats(
-                port_rid,
+        sai_status_t status = sai_metadata_sai_queue_api->get_queue_stats(
+                queue_rid,
                 countersSize,
-                m_supported_counters.data(),
+                m_supported_queue_counters.data(),
                 counters.data());
 
         if (status != SAI_STATUS_SUCCESS)
         {
-            SWSS_LOG_ERROR("failed to collect counters for port RID %s: %s",
-                    sai_serialize_object_id(port_rid).c_str(),
+            SWSS_LOG_ERROR("failed to collect queue counters for queue RID %s: %s",
+                    sai_serialize_object_id(queue_rid).c_str(),
                     sai_serialize_status(status).c_str());
+
             continue;
         }
 
+<<<<<<< HEAD
         sai_object_id_t vid = translate_rid_to_vid(port_rid, m_switch_vid);
 
         std::string strPortId = sai_serialize_object_id(vid);
+=======
```

Comment:
build break?

Reply:
I'm sorry, I've not added the merge fixes to commit. It's already updated.